RTCStatsReport - make this the center of statistics docs (#27028)
mdn/content: files/en-us/web/api/rtcicecandidatepairstats/index.md

title: RTCIceCandidatePairStats
slug: Web/API/RTCIceCandidatePairStats
page-type: web-api-interface
browser-compat: api.RTCStatsReport.type_candidate-pair
---

{{APIRef("WebRTC")}}

The **`RTCIceCandidatePairStats`** dictionary of the [WebRTC API](/en-US/docs/Web/API/WebRTC_API) is used to report statistics that provide insight into the quality and performance of an {{domxref("RTCPeerConnection")}} while connected and configured as described by the specified pair of {{Glossary("ICE")}} candidates.

> The statistics can be obtained by iterating the {{domxref("RTCStatsReport")}} returned by {{domxref("RTCPeerConnection.getStats()")}} until you find a report with the [`type`](#type) of `candidate-pair`.

Author: We do

> ## Instance properties

Author: Will fix these when I go through each dictionary individually

- {{domxref("RTCIceCandidatePairStats.availableIncomingBitrate", "availableIncomingBitrate")}} {{optional_inline}}

>   - : Provides an informative value representing the available inbound capacity of the network by reporting the total number of bits per second available for all of the candidate pair's incoming {{Glossary("RTP")}} streams. This does not take into account the size of the {{Glossary("IP")}} overhead, nor any other transport layers such as {{Glossary("TCP")}} or {{Glossary("UDP")}}.

Author: ```suggestion
  - : Provides a value representing the available inbound capacity of the network by reporting the total number of bits per second available for all of the candidate pair's incoming {{Glossary("RTP")}} streams. This does not take into account the size of the {{Glossary("IP")}} overhead, nor any other transport layers such as {{Glossary("TCP")}} or {{Glossary("UDP")}}.
```

- {{domxref("RTCIceCandidatePairStats.transportId", "transportId")}} {{optional_inline}}
  - : A string that uniquely identifies the {{domxref("RTCIceTransport")}} that was inspected to obtain the transport-related statistics (as found in {{domxref("RTCTransportStats")}}) used in generating this object.

### Common instance properties

The following properties are common to all WebRTC statistics objects.

<!-- RTCStats -->

- {{domxref("RTCIceCandidatePairStats.id", "id")}}
  - : A string that uniquely identifies the object that is being monitored to produce this set of statistics.
- {{domxref("RTCIceCandidatePairStats.timestamp", "timestamp")}}
  - : A {{domxref("DOMHighResTimeStamp")}} object indicating the time at which the sample was taken for this statistics object.
- {{domxref("RTCIceCandidatePairStats.type", "type")}}
  - : A string with the value `"candidate-pair"`, indicating the type of statistics that the object contains.

### Deprecated properties

The following properties have been removed from the specification and should no longer be used.
You should update any existing code to avoid using them as soon as is practical.

> Check the [compatibility table](#browser_compatibility) for details on which browsers support them and in which versions.

Author: I'll leave this open, but do not plan to fix in this PR.
